crates/uv/tests/it/sync.rs

```diff
@@ -7850,15 +7850,326 @@ fn multiple_group_conflicts() -> Result<()> {
+    ");
+
+    Ok(())
+}
+
+/// See: <https://github.com/astral-sh/uv/issues/11232>
+#[test]
+fn transitive_group_conflicts_siblings() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "example"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = []
+
+        [dependency-groups]
+        dev = [
+            { include-group = "test" },
+        ]
+        dev2 = [
+            { include-group = "magic" },
+        ]
+        test = ["tox>4"]
+        magic = ["tox<4"]
+
+        [tool.uv]
+        conflicts = [
+            [
+                { group = "test" },
+                { group = "magic" },
+            ],
+        ]"#,
+    )?;
+
+    uv_snapshot!(context.filters(), context.sync(), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 15 packages in [TIME]
+    Prepared 11 packages in [TIME]
+    Installed 11 packages in [TIME]
+     + cachetools==5.3.3
+     + chardet==5.2.0
+     + colorama==0.4.6
+     + distlib==0.3.8
+     + filelock==3.13.1
+     + packaging==24.0
+     + platformdirs==4.2.0
+     + pluggy==1.4.0
+     + pyproject-api==1.6.1
+     + tox==4.14.2
+     + virtualenv==20.25.1
+    ");
+
+    uv_snapshot!(context.filters(), context.sync().arg("--group").arg("dev"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 15 packages in [TIME]
+    Audited 11 packages in [TIME]
+    ");
+
+    uv_snapshot!(context.filters(), context.sync().arg("--no-dev").arg("--group").arg("dev2"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 15 packages in [TIME]
+    Prepared 3 packages in [TIME]
+    Uninstalled 5 packages in [TIME]
+    Installed 3 packages in [TIME]
+     - cachetools==5.3.3
+     - chardet==5.2.0
+     - colorama==0.4.6
+     + py==1.11.0
+     - pyproject-api==1.6.1
+     + six==1.16.0
+     - tox==4.14.2
+     + tox==3.28.0
+    ");
+
+    uv_snapshot!(context.filters(), context.sync().arg("--group").arg("dev2"), @r"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 15 packages in [TIME]
+    error: Groups `dev` (enabled by default) and `dev2` are incompatible with the transitively inferred conflicts: {`example:dev`, `example:dev2`}
+    ");
+
+    uv_snapshot!(context.filters(), context.sync().arg("--group").arg("dev").arg("--group").arg("dev2"), @r"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 15 packages in [TIME]
+    error: Groups `dev` and `dev2` are incompatible with the transitively inferred conflicts: {`example:dev`, `example:dev2`}
+    ");
```

Comment:
Nice tests!

Reply:
Should we add a test that covers cycle detection?

Reply:
Good idea. Added one!